src/coreclr/src/pal/tests/palsuite/common/palsuite.h

```diff
@@ -17,6 +17,7 @@
 #ifndef __PALSUITE_H__
 #define __PALSUITE_H__
 
+#include <uchar.h>
```

Comment:
@janvorli, this header is only available on Linux. On macOS, Solaris etc. it is not available (re: #34776). It is only available in std=c++11 and onwards context, which we are not using in case of palsuite.

Could palsuite be changed so we don't depend on this non-standard setup:
> system provided `char16_t` definition when std != c++11+

I am struggling to compile this test project on Solaris port, and running the PAL suite is quite essential on new platform bring up.